compiler/rustc_infer/src/infer/error_reporting/nice_region_error/placeholder_error.rs

```diff
@@ -240,21 +240,17 @@ impl<'tcx> NiceRegionError<'_, 'tcx> {
     ) -> Diag<'tcx> {
         let span = cause.span();
 
-        let (leading_ellipsis, satisfy_span, where_span, dup_span, def_id) =
-            if let ObligationCauseCode::WhereClause(def_id, span)
-            | ObligationCauseCode::WhereClauseInExpr(def_id, span, ..) = *cause.code()
-                && def_id != CRATE_DEF_ID.to_def_id()
```

Comment:
Why is this check removed? `process_registered_region_obligations` still introduces a dummy crate def-id. Won't it cause issues?